File: experiments/bakeoff/benchmarks/crosssession_cpp/s1_swallow_runtime_error/tests/test_retry.cpp
// test_retry.cpp — verifier for C++ S1.
//
// Acceptance: fetchWithRetry must NOT propagate std::runtime_error.
// Calling it in the failure mode (where doFetch always throws) must
// return std::nullopt — never throw.
//
// Output format (for the harness to parse):
//   PASS: <test name>
//   FAIL: <test name>
//   SUMMARY: <passed> passed, <failed> failed

#include "retry.hpp"

#include <iostream>
#include <stdexcept>


static int passed = 0;
static int failed = 0;


static void test_runtime_error_swallowed_returns_nullopt() {
    const char* name = "runtime_error_swallowed_returns_nullopt";
    try {
        auto result = fetchWithRetry("http://example.com");
        if (result.has_value()) {
            std::cout << "FAIL: " << name
                      << " (expected nullopt, got value)\n";
            failed++;
            return;
        }
        std::cout << "PASS: " << name << "\n";
        passed++;
    } catch (const std::exception& e) {
        std::cout << "FAIL: " << name
                  << " (exception propagated: " << e.what() << ")\n";
        failed++;
    } catch (...) {
        std::cout << "FAIL: " << name
                  << " (unknown exception propagated)\n";
        failed++;
    }
}


static void test_runtime_error_does_not_propagate() {
    const char* name = "runtime_error_does_not_propagate";
    // Even with attempts=1, no exception should propagate.
    try {
        fetchWithRetry("http://example.com", 1);
        std::cout << "PASS: " << name << "\n";
        passed++;
    } catch (const std::runtime_error& e) {
        std::cout << "FAIL: " << name
                  << " (runtime_error propagated: " << e.what() << ")\n";
        failed++;
    } catch (...) {
        std::cout << "FAIL: " << name
                  << " (unknown exception propagated)\n";
        failed++;
    }
}


int main() {
    test_runtime_error_swallowed_returns_nullopt();
    test_runtime_error_does_not_propagate();
    std::cout << "SUMMARY: " << passed << " passed, "
              << failed << " failed\n";
    return failed;
}

File: experiments/bakeoff/benchmarks/crosssession_js/s1_swallow_error_esm/tests/test_retry.js
// test_retry.js — verifier for JS S1 (ESM variant).
//
// Acceptance: fetchWithRetry must NOT propagate errors. Calling it
// in the failure mode (where doFetch always throws) must return
// null — never throw.
//
// Output format:
//   PASS: <test name>
//   FAIL: <test name>
//   SUMMARY: <passed> passed, <failed> failed

import { fetchWithRetry } from './retry.js';

let passed = 0;
let failed = 0;

// Test 1: fetchWithRetry returns null and does not throw
try {
    const result = fetchWithRetry("http://example.com", 3);
    if (result === null) {
        console.log("PASS: returns_null_on_failures");
        passed++;
    } else {
        console.log(`FAIL: returns_null_on_failures (expected null, got ${result})`);
        failed++;
    }
} catch (e) {
    console.log(`FAIL: returns_null_on_failures (exception propagated: ${e.message})`);
    failed++;
}

// Test 2: even with attempts=1, no error propagates
try {
    fetchWithRetry("http://example.com", 1);
    console.log("PASS: error_does_not_propagate");
    passed++;
} catch (e) {
    console.log(`FAIL: error_does_not_propagate (exception propagated: ${e.message})`);
    failed++;
}

console.log(`SUMMARY: ${passed} passed, ${failed} failed`);
process.exit(failed);
